app/src/main/java/com/beemdevelopment/aegis/ui/EditEntryActivity.java

```diff
@@ -354,36 +362,106 @@ protected void onCreate(Bundle savedInstanceState) {
-                        }
+        _textGroup.setShowSoftInputOnFocus(false);
+        _textGroup.setOnClickListener(v -> showGroupSelectionDialog());
+        _textGroup.setOnFocusChangeListener((v, hasFocus) -> {
+            if (hasFocus) {
+                showGroupSelectionDialog();
+            }
+        });
+
+        _textGroupLayout.setOnClickListener(v -> {
+            showGroupSelectionDialog();
+        });
+
+        _textUsageCount.setText(_prefs.getUsageCount(entryUUID).toString());
+        setLastUsedTimestamp(_prefs.getLastUsedTimestamp(entryUUID));
+    }
 
-                        updateGroupDropdownList();
-                        setGroup(new VaultGroupModel(group));
+    private void showGroupSelectionDialog() {
+        BottomSheetDialog dialog = new BottomSheetDialog(this);
+        View view = getLayoutInflater().inflate(R.layout.dialog_select_groups, null);
+        dialog.setContentView(view);
+
+        ChipGroup chipGroup = view.findViewById(R.id.groupChipGroup);
+        TextView addGroupInfo = view.findViewById(R.id.addGroupInfo);
+        LinearLayout addGroup = view.findViewById(R.id.addGroup);
+        Button clearButton = view.findViewById(R.id.btnClear);
+        Button saveButton = view.findViewById(R.id.btnSave);
+
+        chipGroup.removeAllViews();
+        addGroupInfo.setVisibility(View.VISIBLE);
+        addGroup.setVisibility(View.VISIBLE);
+
+        for (VaultGroup group : _groups) {
+            addChipTo(chipGroup, new VaultGroupModel(group), false);
+        }
+
+        addGroup.setOnClickListener(v1 -> {
+            Dialogs.TextInputListener onAddGroup = text -> {
+                String groupName = new String(text).trim();
+                if (!groupName.isEmpty()) {
+                    VaultGroup group = _vaultManager.getVault().findGroupByName(groupName);
+                    if (group == null) {
+                        group = new VaultGroup(groupName);
+                        _vaultManager.getVault().addGroup(group);
                     }
-                };
 
-                DialogInterface.OnCancelListener onCancel = dialogInterface -> {
-                    VaultGroupModel previous = (VaultGroupModel) _dropdownGroup.getTag();
-                    _dropdownGroup.setText(previous.getName(), false);
-                };
+                    _selectedGroups.add(group.getUUID());
+                    addChipTo(chipGroup, new VaultGroupModel(group), true);
+                }
+            };
 
-                Dialogs.showTextInputDialog(EditEntryActivity.this, R.string.set_group, R.string.group_name_hint, onAddGroup, onCancel);
+            Dialogs.showTextInputDialog(EditEntryActivity.this, R.string.set_group, R.string.group_name_hint, onAddGroup, null);
+        });
+
+        saveButton.setOnClickListener(v1 -> {
+            if(getCheckedUUID(chipGroup).isEmpty()) {
+                _textGroup.setText(getString(R.string.no_group));
```

Comment:
I think we're missing ``_selectedGroups.clear();`` here. Otherwise, it's not possible to remove an entry from all groups.

Reply:
Missed that !!